Makefile

```diff
@@ -70,6 +70,43 @@ run-docker:
 		-p 8080:8080 \
 		hypothesis/h-periodic:$(DOCKER_TAG)
 
+# Tell make how to compile requirements/*.txt files.
+#
+# `touch` is used to pre-create an empty requirements/%.txt file if none
+# exists, otherwise tox crashes.
+#
+# $(subst) is used because in the special case of making requirements.txt we
+# actually need to touch dev.txt not requirements.txt and we need to run
+# `tox -e dev ...` not `tox -e requirements ...`
+#
+# $(basename $(notdir $@))) gets just the environment name from the
+# requirements/%.txt filename, for example requirements/foo.txt -> foo.
+requirements/%.txt: requirements/%.in
+	@touch -a $(subst requirements.txt,dev.txt,$@)
+	@tox -qe $(subst requirements,dev,$(basename $(notdir $@))) --run-command 'pip --quiet --disable-pip-version-check install pip-tools'
+	@tox -qe $(subst requirements,dev,$(basename $(notdir $@))) --run-command 'pip-compile --allow-unsafe --quiet $(args) $<'
+
+# Inform make of the dependencies between our requirements files so that it
+# knows what order to re-compile them in and knows to re-compile a file if a
+# file that it depends on has been changed.
+requirements/dev.txt: requirements/requirements.txt
+requirements/tests.txt: requirements/requirements.txt
+requirements/functests.txt: requirements/requirements.txt
+requirements/lint.txt: requirements/tests.txt requirements/functests.txt
+
+# Add a requirements target so you can just run `make requirements` to
+# re-compile *all* the requirements files at once.
+#
+# This needs to be able to re-create requirements/*.txt files that don't exist
+# yet or that have been deleted so it can't just depend on all the
+# requirements/*.txt files that exist on disk $(wildcard requirements/*.txt).
+#
+# Instead we generate the list of requirements/*.txt files by getting all the
+# requirements/*.in files from disk ($(wildcard requirements/*.in)) and replace
+# the .in's with .txt's.
+.PHONY: requirements requirements/
+requirements requirements/: $(foreach file,$(wildcard requirements/*.in),$(basename $(file)).txt)
```

Comment:
This adds the `make requirements` command for (re-)compiling `requirements/*.txt`, same as `h-periodic` and all our other projects have.